hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/consumer/sender/ProtocolMessageSenderProvider.java

```diff
@@ -1,10 +1,18 @@
 package pl.allegro.tech.hermes.consumers.consumer.sender;
 
 import pl.allegro.tech.hermes.api.EndpointAddress;
+import pl.allegro.tech.hermes.api.Subscription;
 
 public interface ProtocolMessageSenderProvider {
 
-    MessageSender create(EndpointAddress endpoint);
+    @Deprecated
+    default MessageSender create(EndpointAddress endpoint) {
+        throw new UnsupportedOperationException();
+    }
+
+    default MessageSender create(Subscription subscription) {
+        return create(subscription.getEndpoint());
```

Comment:
i don't think having two default methods is a good idea. i might not implement any of them and it will be fine for the compiler. If we are looking at backwards compatibility, we can only make the new method default, not the old one i guess?